documentation/modules/ROOT/partials/modules/mysql-connector/con-mysql-connector-events.adoc

```diff
@@ -3,15 +3,67 @@
 [id="mysql-connector-events_{context}"]
 = MySQL connector events
 
-All data change events produced by the {prodname} MySQL connector contain a key and a value. The change event key and the change event value each contain a _schema_ and a _payload_ where the schema describes the structure of the payload and the payload contains the data.
+The {prodname} MySQL connector generates a data change event for each row-level `INSERT`, `UPDATE`, and `DELETE` operation. Each event contains a key and a value. The structure of the key and the value depends on the table that was changed. 
 
-WARNING: The MySQL connector ensures that all Kafka Connect schema names adhere to the link:http://avro.apache.org/docs/current/spec.html#names[Avro schema name format]. This is important as any character that is not a latin letter or underscore is replaced by an underscore which can lead to unexpected conflicts in schema names when the logical server names, database names, and table names container other characters that are replaced with these underscores.
+{prodname} and Kafka Connect are designed around _continuous streams of event messages_. However, the structure of these events may change over time, which can be difficult for consumers to handle. To address this, each event contains the schema for its content, which makes each event self-contained. The following skeleton JSON shows the basic structure of a change event: 
 
-== Change event key
+[source,json,index=0]
+----
+{
+ "schema": { //<1>
```

Comment:
Note that "schema" will only be present if the JSON converter is configured to produce it. It also can only emit the actual key and value payloads optionally. We should somehow make clear that this is more a conceptual representation, which can look different depending on chosen configuration (JSON with or without schema, JSON with schema in the registry when using Apicurio, Avro instead of JSON, or even yet another format like ProtoBuf...). 

Perhaps say somewhere/somehow that those four parts key/value schema/payload exist, but it's a matter of the chosen converter how they are represented in actual change events on the wire. And then focus on the JSON case with schema as the example.

Reply:
The skeleton JSON now has this introductory paragraph: 

The following skeleton JSON shows the basic four parts of a change event. However, how you configure the Kafka Connect converter that you choose to use in your application determines the representation of these four parts in change events. A `schema` field is in a change event only when you configure the converter to produce it. Likewise, the event key and event payload are in a change event only if you configure a converter to produce it. If you use the JSON converver and you configure it to produce all four basic change event parts, change events have this structure: 

Is that correct, clear, complete? Let me know if something needs to change.

Reply:
@gunnarmorling - is this correct?

Reply:
@TovaCohen What you have sounds fine and expresses what I believe Gunnar was after.